src/Mod/PartDesign/Gui/TaskChamferParameters.cpp

```diff
@@ -194,6 +198,31 @@ void TaskChamferParameters::clearButtons(const selectionModes notThis)
     DressUpView->highlightReferences(false);
 }
 
+// By adding the edges to the Gui.Selection in python, then onSelectionChanged() gets called
+// where the edges get added to the list in the task dialog and highlighted in the 3d view
+
+void TaskChamferParameters::onSelectAllEdges(void)
+{
+    PartDesign::Chamfer* pcChamfer = static_cast<PartDesign::Chamfer*>(DressUpView->getObject());
+    if (pcChamfer) {
+        App::DocumentObject* base = pcChamfer->Base.getValue();
+        if (base) {
+            std::string baseName = base->getNameInDocument();
+            std::ostringstream command;
+            command << "from PySide import QtGui,QtCore\n";
+            command << "QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)\n";

```

Comment:
I wonder whether it's a good idea to put the override cursor into the macro code. In case an exception is raised before restoring the old cursor the interpreter jumps out of the function and will never restore the old cursor.

Actually it should be sufficient to only show the wait cursor if the dialog is opened and thus it can be done in C++ by using the class Gui::WaitCursor.